ui/tests/unit/abilities/variable-test.js

```diff
@@ -15,87 +15,151 @@ module('Unit | Ability | variable', function (hooks) {
+          {
+            rulesJSON: {
+              Namespaces: [
+                {
+                  Name: 'default',
+                  Capabilities: [],
+                  SecureVariables: {
+                    'Path "*"': {
+                      Capabilities: ['list'],
+                    },
                   },
                 },
-              },
-            ],
+              ],
+            },
           },
-        },
-      ],
+        ],
+      });
+
+      this.owner.register('service:token', mockToken);
+
+      assert.ok(this.ability.canList);
     });
 
-    this.owner.register('service:token', mockToken);
+    test('it permits listing variables when token has SecureVariables alone in its rules', function (assert) {
+      const mockToken = Service.extend({
+        aclEnabled: true,
+        selfToken: { type: 'client' },
+        selfTokenPolicies: [
+          {
+            rulesJSON: {
+              Namespaces: [
+                {
+                  Name: 'default',
+                  Capabilities: [],
+                  SecureVariables: {},
+                },
+              ],
+            },
+          },
+        ],
+      });
+
+      this.owner.register('service:token', mockToken);
 
-    assert.ok(this.ability.canList);
+      assert.ok(this.ability.canList);
+    });
   });
 
-  test('it permits listing variables when token has SecureVariables alone in its rules', function (assert) {
-    const mockToken = Service.extend({
-      aclEnabled: true,
-      selfToken: { type: 'client' },
-      selfTokenPolicies: [
-        {
-          rulesJSON: {
-            Namespaces: [
-              {
-                Name: 'default',
-                Capabilities: [],
-                SecureVariables: {},
-              },
-            ],
-          },
-        },
-      ],
+  module('#create', function () {
+    test('it does not permit creating variables by default', function (assert) {
+      const mockToken = Service.extend({
+        aclEnabled: true,
+      });
+
+      this.owner.register('service:token', mockToken);
+
+      assert.notOk(this.ability.canCreate);
+    });
+
+    test('it permits creating variables when token type is management', function (assert) {
+      const mockToken = Service.extend({
+        aclEnabled: true,
+        selfToken: { type: 'management' },
+      });
+
+      this.owner.register('service:token', mockToken);
+
+      assert.ok(this.ability.canCreate);
+    });
+
+    test('it permits creating variables when acl is disabled', function (assert) {
+      const mockToken = Service.extend({
+        aclEnabled: false,
+        selfToken: { type: 'client' },
+      });
+
+      this.owner.register('service:token', mockToken);
+
+      assert.ok(this.ability.canCreate);
     });
 
-    this.owner.register('service:token', mockToken);
+    test('it permits creating variables when token has SecureVariables with create capabilities in its rules', function (assert) {
```

Comment:
Something to consider on these tests is that we don't test many of the "it doesn't permit" cases. The one that jumps out at me is that we don't test that having a capability in one namespace doesn't give access to that same capability in another namespace.

Reply:
Yup. Namespaces are a behemoth of an issue and one of the most important testing stories for the UI. This is ticketed in [13310](https://github.com/hashicorp/nomad/issues/13310#issuecomment-1159178019) which this PR is a prerequisite to.

We should totally collaborate on creating a list of plain English specifications around namespaces before we get too deep into implementation. Which is what I had in mind [yesterday](https://hashicorp.slack.com/archives/C033DJD6EJJ/p1655743562030679?thread_ts=1655742096.654959&cid=C033DJD6EJJ).

Great job staying active on these PRs by the way!! Looking forward to collaborating with you more.